tests/test_shapes.py

```diff
@@ -105,62 +106,89 @@ def test_shapes():
-        resolution=20,
+    # Act
+    shape = tpms.generate(type_part=type_part, smoothing=0, verbose=False)
+    shape_check = tpms.generateVtk(type_part=type_part)
+
+    # Assert
+    assert np.isclose(shape.Volume(), np.abs(shape_check.volume), rtol=1e-2)
+
+
+@pytest.mark.parametrize("type_part", ["lower skeletal", "upper skeletal"])
+def test_tpms_given_cadquery_vtk_zero_offset_skeletals_volume_must_be_equivalent(type_part: str):
+    # Arrange
+    tpms = microgen.Tpms(
+        surface_function=microgen.surface_functions.schwarzP,
+        offset=0.0,
     )
-    # elem.generate(type_part="surface")
-    _ = elem.upper_skeletal
-    _ = elem.upper_skeletal
-    _ = elem.generateVtk(type_part="sheet")
 
-    elem = microgen.shape.tpms.CylindricalTpms(
-        radius=1.0,
-        surface_function=microgen.shape.surface_functions.schwarzD,
-        repeat_cell=(1, 10, 1),
+    # Act
+    shape = tpms.generate(type_part=type_part, smoothing=0, verbose=False)
+    shape_check = tpms.generateVtk(type_part=type_part)
+
+    # Assert
+    assert np.isclose(shape.Volume(), np.abs(shape_check.volume), rtol=1e-2)
+
+@pytest.mark.parametrize("surface", [func[0] for func in getmembers(microgen.surface_functions, isfunction)])
+def test_tpms_given_sum_volume_must_be_cube_volume(surface: str):
+    # Arrange
+    tpms = microgen.Tpms(
+        surface_function=getattr(microgen.surface_functions, surface),
+        offset=1.0,
+        repeat_cell=2,
+        cell_size=3.0,
     )
-    _ = elem.lower_skeletal
-    _ = elem.lower_skeletal
 
-    elem = microgen.shape.tpms.SphericalTpms(
+    # Act
+    volume = np.abs(tpms.sheet.volume + tpms.lower_skeletal.volume + tpms.upper_skeletal.volume)
+    cube_volume = np.prod(tpms.repeat_cell) * np.prod(tpms.cell_size)
+
+    # Assert
+    assert np.isclose(volume, cube_volume, rtol=1e-2)
+
+@pytest.mark.parametrize("coord_sys_tpms", [microgen.CylindricalTpms, microgen.SphericalTpms])
+def test_tpms_given_coord_system_tpms_volumes_must_be_greater_than_zero_and_lower_than_grid_volume(coord_sys_tpms):
+    tpms = coord_sys_tpms(
         radius=1.0,
-        surface_function=microgen.shape.surface_functions.schwarzD,
-        repeat_cell=(1, 10, 10),
+        surface_function=microgen.surface_functions.gyroid,
+        offset=1.0,
     )
-    _ = elem.surface
-    _ = elem.surface
 
+    assert 0 < tpms.sheet.extract_surface().volume < np.abs(tpms.grid.volume)
+    assert 0 < tpms.lower_skeletal.extract_surface().volume < np.abs(tpms.grid.volume)
+    assert 0 < tpms.upper_skeletal.extract_surface().volume < np.abs(tpms.grid.volume)
+
+def test_tpms_given_variable_thickness_cadquery_and_vtk_volumes_must_correspond():
+    def graded_density(x, y, z):
+        return x + 1.5
+
+    tpms = microgen.Tpms(
+        surface_function=microgen.surface_functions.gyroid,
+        offset=graded_density,
+    )
+
+    cadquery_part = tpms.generate(type_part="sheet", smoothing=0, verbose=False)

```

Comment:
Try to keep consistent variable names, `shape` and `shape_check` or `cadquery_part` and `vtk_part` but not both.